modules/artifact/src/main/java/org/mule/module/artifact/classloader/ClassLoaderResourceReleaser.java

```diff
@@ -67,45 +61,4 @@ private void clearClassLoaderSoftkeys() {
       logger.warn("Couldn't clear soft keys in caches. This can cause a classloader memory leak.", e);
     }
   }
-
-  /**
-   * Shutdowns the AWS IdleConnectionReaper Thread if one is present, since it will cause a leak if not closed correctly.
-   */
-  private void shutdownAwsIdleConnectionReaperThread() {
-
-    Class<?> idleConnectionReaperClass;
-    try {
-      idleConnectionReaperClass = this.classLoader.loadClass("com.amazonaws.http.IdleConnectionReaper");
-      try {
-        Method registeredManagersMethod = idleConnectionReaperClass.getMethod("getRegisteredConnectionManagers");
-        List<Object> httpClientConnectionManagers = (List<Object>) registeredManagersMethod.invoke(null);
-        if (httpClientConnectionManagers.isEmpty()) {
-          return;
-        }
-
-        Class<?> httpClientConnectionManagerClass =
-            this.classLoader.loadClass("org.apache.http.conn.HttpClientConnectionManager");
-        Method removeConnectionManager =
-            idleConnectionReaperClass.getMethod("removeConnectionManager", httpClientConnectionManagerClass);
-        for (Object connectionManager : httpClientConnectionManagers) {
-          boolean removed = (boolean) removeConnectionManager.invoke(null, connectionManager);
-          if (!removed && logger.isDebugEnabled()) {
-            logger
-                .debug(format("Unable to unregister HttpClientConnectionManager instance [%s] associated to AWS's IdleConnectionReaperThread",
-                              connectionManager));
-          }
-        }
-
-      } finally {
-        Method shutdown = idleConnectionReaperClass.getMethod("shutdown");
-        shutdown.invoke(null);
-      }
-
-    } catch (ClassNotFoundException | NoSuchMethodException | IllegalArgumentException e) {
-      // If the class or method is not found, there is nothing to dispose
-    } catch (SecurityException | IllegalAccessException | InvocationTargetException e) {
-      logger.warn("Unable to shutdown AWS's IdleConnectionReaperThread, an error occurred: " + e.getMessage(), e);
-    }
-  }
-
```

Comment:
This part has been extracted to its own Releaser, so we can have what is core separated from what is connector-specific.